components/locid_transform/src/fallback/algorithms.rs

```diff
@@ -38,28 +58,31 @@ impl<'a> LocaleFallbackerWithConfig<'a> {
                 );
             }
         }
-        // 2. Remove the script if it is implied by the other subtags
-        if let Some(script) = locale.script() {
-            let default_script = self
-                .likely_subtags
-                .l2s
-                .get_copied(&language.into_tinystr().to_unvalidated())
-                .unwrap_or(DEFAULT_SCRIPT);
-            if let Some(region) = locale.region() {
-                if script
-                    == self
-                        .likely_subtags
-                        .lr2s
-                        .get_copied_2d(
-                            &language.into_tinystr().to_unvalidated(),
-                            &region.into_tinystr().to_unvalidated(),
-                        )
-                        .unwrap_or(default_script)
-                {
+        // 2. Remove the script if it is implied by the other subtags and the fallback priority is
+        // not transliteration.
+        if self.config.priority != FallbackPriority::Transliteration {
```

Comment:
Thought/Suggestion: Rather than _skipping_ this step, it seems like we should instead maximize the script at this point. Then, for example, `ru-RU` becomes `ru-Cyrl-RU` and we run the fallback chain from that reference point. If you do this then you don't need the `max_script` function; just change the body of this branch to add instead of remove the script.

Reply:
`ru-Cyrl-RU` is not part of `ru-RU`'s fallback chain, neither is `ru-Cyrl` that would be the next step: https://unicode.org/reports/tr35/tr35-general.html#Inheritance 

We need to store the maximized script outside the locale itself, because it's invalid to maximize after we've already changed the locale. It needs to be on the initial locale, otherwise e.g. `az` would fallback to `und-Latn`, which is the maximized script for `az` but not for `az-IR`.

I pushed an alternative version that reuses the `normalize` code block.

Reply:
Agreed, ICU4C does what I'm doing AFAICT, though

Reply:
(I checked that supplemental data is not accessed and that the fallback starts from the initial locale, not the maximized one)